### DISCUSSION TITLE
Filtering table from value in formula

### DISCUSSION THREAD

**Discussion by GuilhermeHirsch** [Budibase/budibase#2014] [Help]
Is it possible to filter a table by referencing a formula value? I tried but the rows didn't appear.

![Sem título](https://user-images.githubusercontent.com/33108596/126000953-3083c204-4701-4d1f-8202-e4cd843c6436.jpg)

"ID COBRANÇA" is a formula that references the "auto id" of the relational table.
I noticed that the "equal" operator doesn't appear with the formula.

**Comment by mjashanks** [comment]
Hey @GuilhermeHirsch.

Formula fields cannot be used in filters. The bug here is that the field should not be available for selection in filters. 

If you are interested, this is because formula fields are calculated client-side - in the browser. Filters happen server-side, by the budibase web server - inside a docker container.

Could you apply the filter directly to the Auto ID column, with a handlebars expression? Happy to help if you can describe what you are trying to do.

**Reply by GuilhermeHirsch** [reply]
Helo, @mjashanks! Good Morning!

I have two related tables "Bill" (Cobranças) and "Services Provided" (Serviços Prestados) (A bill has multiple services provided).
My idea is to have on this second screen (clicked on the "Ver" button in the billing table) the possibility of updating the billings and, below, the table of services provided corresponding to this list.
For this, I created a formula in the table of services rendered referencing the Auto ID of the Billings table.
However, as you said, that way is not possible.

![1](https://user-images.githubusercontent.com/33108596/126147379-44201024-f4a9-4e2d-8216-b7c3a050dc7c.jpg)


![2](https://user-images.githubusercontent.com/33108596/126147404-bca4e3b2-5619-472d-84be-90a383b52be0.jpg)

**Reply by mjashanks** [reply]
It sounds like you need to use a one to many relationship between "Cobranças" (one) and "Serviços Prestados" (many).

in the Cobranças edit screen, you can
1. Add a data provider inside the form
2. Set the datasource to `Relationships > Serviços Prestados`
3. Add a table underneath

This will make the table list all "Serviços Prestados" that belong to the current "Cobranças"

**Comment by oswaldB** [comment]
Hello,

I may face a problem in the same veine as yours:
why this filter is not working?
![image](https://user-images.githubusercontent.com/3023526/126153437-8e92da3b-ae30-42d9-8620-8541cb4111d5.png)
"code document" is a custom formula {{random 999 99999 }} that is generated in an automation as un update after a row creation. So code document is a text field.

**Reply by mike12345567** [reply]
Hi @oswaldB - this should work, have you checked your internal table to make sure the `id` value that is in the URL is within the table as you expect?

**Reply by oswaldB** [reply]
Well code document is not the id. It is a custom Id I created with an update after the row creation thanks to an automation.

So to be clear:
documents/12345 => URL.id = 12345 and the filter will works only if I have an document with an Id = 12345 only. 

I cannot use {{url.i}} to filter on another column than the id?

**Reply by oswaldB** [reply]
![image](https://user-images.githubusercontent.com/3023526/126155950-09144850-6386-4aa4-8b70-cf712ceddd4d.png)
my data if it can help you

**Reply by mike12345567** [reply]
Hi @oswaldB - you could change how you link into the screen, by default the `URL.id` will contain the value of the `_id` field, but depending on where you are pushing the user from you could change this, for example if you're using a link then you can change the URL to use `code document` rather than the `_id` field, or if you're using a button in the button action you can change the URL it is linking to.

Here you can see a link, I've opened up the binding to change it.
![image](https://user-images.githubusercontent.com/4407001/126157039-438c3ce8-69be-40e7-9559-ede5c53cf8d9.png)

**Reply by oswaldB** [reply]
It is what I did:
![image](https://user-images.githubusercontent.com/3023526/126159664-c5337251-631e-44e5-afdc-2ee3b6ada90b.png)

My question is how can I set up the filter to show only the entry with this code document?

**Reply by oswaldB** [reply]
the problem is on the filter and not in the transfer of information. because I can set up  a text that equals {{url.id}} and it works. it shows 12345 in my case.

**Reply by GuilhermeHirsch** [reply]
Same situation with me, @oswaldB.

**Reply by oswaldB** [reply]
![image](https://user-images.githubusercontent.com/3023526/126161296-972b93e0-fd6c-4d24-ad1d-87b1b0dc4d48.png)
this is not working.

**Reply by mike12345567** [reply]
Just looking into this, there seems to be an issue with the equality operator in this scenario, looking into it (it appears to function as expected when using the starts with, but that isn't the right operator for this scenario).

**Reply by GuilhermeHirsch** [reply]
Exactly, @mike12345567. With the "starts with" operator here it worked too. But with the "equality" operator no.

**Reply by oswaldB** [reply]
I confirm it works with the "start with"

**Reply by mike12345567** [reply]
I've put up PR to attempt to fix the issues here, PR found here: https://github.com/Budibase/budibase/pull/2022

**Reply by GuilhermeHirsch** [reply]
A very beginner's question, @mike12345567: This code change will be reflected in the next update, right?

**Reply by mike12345567** [reply]
Hi @GuilhermeHirsch - it'll be released once its merged into the master branch, which should be very soon - we will release this as a hotfix.

**Comment by oswaldB** [comment]
Thank you :)
